node_scripts/submit.py

```diff
@@ -162,18 +162,32 @@ def recieve_submit_request(application_file_path):
         driver_cores=application['driver_cores'],
         executor_cores=application['executor_cores'])
 
-    try:
-        return_code = subprocess.call(cmd.to_str(), shell=True)
-        upload_log(blob_client, application)
-        return return_code
-    except subprocess.CalledProcessError as e:
-        print(e)
-        upload_log(blob_client, application)
-        return 1
+    return_code = subprocess.call(cmd.to_str(), shell=True)
+    upload_log(blob_client, application)
+    return return_code
+
+
+def upload_error_log(error, application_file_path):
```

Comment:
Should we _always_ upload these? Maybe an option to override at job submit?

Reply:
I think we should always upload on an error. If you don't there is no explicit way to tell that something went wrong, or what went wrong. 

On a successful run, error logs won't be uploaded. Note that a successful run includes when a spark application has failed. An error here is only when something goes wrong on AZTK's end.